src/plugins/syntax-highlighting/index.ts

```diff
@@ -0,0 +1,96 @@
+import { PluginController } from "../PluginController";
+import { generateBracketPairColorizationCSS } from "./bracket-pair-colorization";
+import { Config, configList } from "./config";
+import "./index.less";
+
+// assumes valid input;
+export function hex2rgb(hex: string): [number, number, number] {
+  return [
+    parseInt(hex.slice(1, 3), 16),
+    parseInt(hex.slice(3, 5), 16),
+    parseInt(hex.slice(5, 7), 16),
+  ];
+}
+
+export function toggleRules(rules: string[], toggle: boolean) {
+  return toggle ? rules : [];
+}
+
```

Comment:
This function seems unused.

Reply:
ah, the good 'ol speculative abstraction

Reply:
fixed